sound/soc/sof/ipc4-topology.c

```diff
@@ -556,6 +558,11 @@ static void sof_ipc4_widget_free_comp_dai(struct snd_sof_widget *swidget)
 	if (!dai)
 		return;
 
+	if (!dai->private) {
+		kfree(dai);
```

Comment:
You might want to `swidget->private = NULL;` here as well.

Or add a label and jump to the end of the function.

Reply:
Right. Will do it.